libraries/AP_Airspeed/AP_Airspeed_ASP5033.cpp

```diff
@@ -0,0 +1,175 @@
+#define REG_WAI_SET            0XA4
+#define REG_SENSOR_READY       0x08
+#define REG_WHOAMI_DEFAULT_ID  0X00
+#define REG_WHOAMI_RECHECK_ID  0X66
+#define CMD_MEASURE            0x0A
+#define RAW_DATA_ON            0b11111101
+
+
+
+bool AP_Airspeed_ASP5033::init()
+{
+    // probe the sensor
+    const uint8_t addresses[] = { ASP5033_I2C_ADDR_1, ASP5033_I2C_ADDR_2 };
+    for (uint8_t i=0; i<ARRAY_SIZE(addresses); i++) {
+        _dev = hal.i2c_mgr->get_device(get_bus(), addresses[i]);
+       if (!_dev && i <ARRAY_SIZE(addresses)) {
+         continue;
+       }
+       if(!_dev){
+         return false;
+       }
+    }
+    
+    WITH_SEMAPHORE(_dev->get_semaphore());
+    _dev->set_speed(AP_HAL::Device::SPEED_HIGH);
+
+    //check WhoamI
+    uint8_t checkwhoami;
+    if (!_dev->read_registers(REG_WAI_SET, &checkwhoami, 1) || //Read the whoami ID
+        checkwhoami != REG_WHOAMI_DEFAULT_ID){
+        _dev->set_retries(2);
+        return false;      
+    }
+
+    //Confirm it's ASP5033 sensor, not the other.
+    bool confirm = false;
+    confirm = Sensor_confirm();
+    if (!confirm) {
+        return false;
+    }
+
+    WITH_SEMAPHORE(sem);
+    uint8_t SysID;       
+    if (_dev->read_registers(REG_SYS_CONFIG, &SysID, 1)|| //Read Sensor_config
+        SysID != (SysID | RAW_DATA_ON)){ //Read the 0xA5 register value, put the read binary value "and" on "11111101" then write to 0xA5.
+    _dev->write_register(REG_SYS_CONFIG, SysID);     //Set Sensor_config
+    }
+    _dev->register_periodic_callback(1000000UL/80U,
+                                    FUNCTOR_BIND_MEMBER(&AP_Airspeed_ASP5033::timer, void));
+    return true;
+}
+
+bool AP_Airspeed_ASP5033::Sensor_confirm(void)
+{
+    //Confirm it's ASP5033 sensor, not the other.
+    //Set Whoami ID. 0XA4 is a OPT WhoamI Part ID register, also can be read from address 0x01.
+    //check whoami, set the PartID register to 0X66 first, them Read the PartID register.
+    uint8_t whoami;
+    _dev->write_register(REG_WAI_SET, REG_WHOAMI_RECHECK_ID);
+    if (!_dev->read_registers(REG_WAI, &whoami, 1) || //Read the whoami ID
+        whoami != REG_WHOAMI_RECHECK_ID){
+        return false; // it maybe not a ASP5033, tuneoff ASP5033
+    }   
+    else{
+        return true;  
+    }    
+}
+
+
+// read the data from the sensor
+void AP_Airspeed_ASP5033::timer()
+{
+    _dev->write_register(REG_CMD, CMD_MEASURE); //once temperature conversion immediately followed by once sensor signal conversion
+    uint8_t SoC; //Start of conversion flag
+    bool sensing = true;    //sensor m
+    //uint8_t Ready = 0x01;
+    if (_dev->read_registers(REG_CMD, &SoC, 1)){ //Judge whether Data collection is over
+    if ((SoC & REG_SENSOR_READY) != sensing){   // Read the 0x30 register address. If Sco bit is 0,signify the acquisition end, the data can be read.
+        uint8_t pressraw_bytes[2]; //pressure data AD value       
+        if(_dev->read_registers(REG_PRESS_DATA, pressraw_bytes, 2)){
```

Comment:
I've got the datasheet now, it looks like we should be doing a single 5 byte read, giving 3 bytes of pressure and 2 bytes of temperature?